src/game/logic/scriptengine/scriptlist.cpp

```diff
@@ -243,19 +243,24 @@ int ScriptList::Get_Read_Scripts(ScriptList **scripts)
  */
 bool ScriptList::Parse_Script_List_Chunk(DataChunkInput &input, DataChunkInfo *info, void *data)
 {
-    captainslog_dbgassert(
-        static_cast<ScriptListReadInfo *>(data)->num_lists < MAX_LIST_COUNT, "Attempting to parse too many script lists.");
+    // The function body has been marginally rearranged to improve readability in regards to the list index and list count.
```

Comment:
I just really think these comments are useless. Adding temporary variables doesn't affect anything. And now this code is very confusing. The list index isn't an index until you go to allocate. L257 should be where the index temp variable is created prior to the increment. `const auto list_index = ...->num_lists++;`

Reply:
What is confusing?

Reply:
The confusing aspect is comparing an index to the max count. That is confusing. You should only compare totals with the max count. What happens in the future if we use a different indexing strategy like some sort of map. Then there is the whole aspect that you shouldn't have index's of invalid entries. It's just asking for trouble to make a potentially invalid index when you could just move the index creation down past the if and we now know that it is valid.

Reply:
Hmm. It's odd that this would be confusing for you. It's like comparing an index to the size of a vector, array. It's most plausible to check if our desired insertion index reached the max or not.
```c++
for (size_t index = 0; index < MyVector.size(); ++index)
   ...
```

That is why you see in some functions also check like this:
```c++
if (index >= MyVector.size())
    return;
```

Reply:
I can improve the comments to make the function purpose easier to understand.

Reply:
New comments added.